src/controller.ts

```diff
@@ -71,6 +79,116 @@ export default function (
       }
     });
 
+    fastify.post<{ Body: { name: string; email: string } }>(
+      '/create-customer',
+      {
+        schema: {
+          body: {
+            type: 'object',
+            required: ['email', 'name'],
+            properties: { name: { type: 'string' }, email: { type: 'string' } },
+          },
+        },
+      },
+      async (req, res) => {
+        const { name, email } = req.body;
+
+        if (!email) {
+          return res.status(404).send({
+            message: 'Email should be provided',
+          });
+        }
+
+        try {
+          const createdCustomer = await paymentService.createCustomer({
+            name,
+            email,
+          });
+
+          return res.send({
+            customerId: createdCustomer.id,
+          });
+        } catch (err) {
+          return res.status(500).send({
+            message: 'Internal Server Error',
+          });
+        }
+      },
+    );
+
+    fastify.get<{
+      Querystring: { email: string };
+      schema: {
+        querystring: {
+          type: 'object';
+          properties: { email: { type: 'number' } };
+        };
+      };
+    }>('/get-customer-id', async (req, res) => {
+      const { email } = req.query;
+
+      if (!email) {
+        return res.status(404).send({
+          message: 'Email should be provided',
+        });
+      }
+
+      try {
+        const { id } = await paymentService.getCustomerIdByEmail(email);
+        return res.status(200).send({
+          customerId: id,
+        });
+      } catch (error) {
+        return res.status(404).send({
+          message: 'CustomerId not found',
+        });
+      }
+    });
+
+    fastify.post<{ Body: { customerId: string; priceId: string; promoCodeId: string } }>(
+      '/create-subscription',
+      {
+        schema: {
+          body: {
+            type: 'object',
+            required: ['customerId', 'priceId'],
+            properties: {
+              customerId: {
+                type: 'string',
+              },
+              priceId: {
+                type: 'string',
+              },
+              promoCodeId: {
+                type: 'string',
+              },
+            },
+          },
+        },
+      },
+      async (req, res) => {
+        const { customerId, priceId, promoCodeId } = req.body;
+
+        try {
+          const subscriptionSetUp = await paymentService.createSubscription(customerId, priceId, promoCodeId);
```

Comment:
Send a signed token on `createCustomer` with the customerId inside so we can ensure that the user doing the request owns the customer id that is provided to subscribe